kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -0,0 +1,681 @@
+     */
+    virtual inline double GetDeltaTime()
+    {
+        return GetModelPart().GetProcessInfo().GetValue(DELTA_TIME);
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+private:
+    ///@}
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart &mrModelPart;
+
+    bool mMoveMeshFlag;
+
+    bool mInitializeWasPerformed = false;
+
+    bool mInitializeSolutionStepWasPerformed = false;
+
+    ExplicitBuilderAndSolverPointerType mpExplicitBuilderAndSolver = nullptr;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief Initialize the DOF set values
+     * This method initializes the current value of the unknown variables in the DOF set
+     */
+    void InitializeDofSetValues()
+    {
+        // Initialize the DOF values
+        auto& r_dof_set = mpExplicitBuilderAndSolver->GetDofSet();
+        block_for_each(
+            r_dof_set,
+            [](DofType& rDof){
+                auto &r_value = rDof.GetSolutionStepValue();
+                r_value = 0.0;
+            }
+        );
+    }
+
+    /**
+     * @brief Auxiliary call to the Initialize()
+     * For a given container, this calls the Initialize() method
+     * @tparam TContainerType Container type template (e.g. elements, conditions, ...)
+     * @param rContainer Reference to the container
+     */
+    template <class TContainerType>
+    void InitializeContainer(TContainerType &rContainer)
+    {
+        const auto& r_process_info = GetModelPart().GetProcessInfo();
+        block_for_each(
+            rContainer,
+            [&](typename TContainerType::value_type& rEntity){rEntity.Initialize(r_process_info);}
+        );
+    }
+
+    /**
+     * @brief Auxiliary call to the InitializeSolutionStep()
+     * For a given container, this calls the InitializeSolutionStep() method
+     * @tparam TContainerType Container type template (e.g. elements, conditions, ...)
+     * @param rContainer Reference to the container
+     */
+    template <class TContainerType>
+    void InitializeSolutionStepContainer(TContainerType &rContainer)
+    {
+        const auto& r_process_info = GetModelPart().GetProcessInfo();
+        block_for_each(
+            rContainer,
+            [&](typename TContainerType::value_type& rEntity){rEntity.InitializeSolutionStep(r_process_info);}
+        );
+    }
+
+    /**
+     * @brief Auxiliary call to the FinalizeSolutionStep()
+     * For a given container, this calls the FinalizeSolutionStep() method
+     * @tparam TContainerType Container type template (e.g. elements, conditions, ...)
+     * @param rContainer Reference to the container
+     */
+    template <class TContainerType>
+    void FinalizeSolutionStepContainer(TContainerType &rContainer)
```

Comment:
actually those functions are already implemented in the entities-utilites:
https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/entities_utilities.h

Reply:
I don't think so... Only the `Initialize` is implemented in the `EntitiesUtilities`. Maybe the entities utilities can be enhanced in another PR to do this operations.

As a comment aside I would like to add that I'm not sure if the approach in the `EntitiesUtilities` to only initialize the `ACTIVE` entities is robust enough since the "activeness" may change as the problem evolves.

Reply:
> As a comment aside I would like to add that I'm not sure if the approach in the EntitiesUtilities to only initialize the ACTIVE entities is robust enough since the "activeness" may change as the problem evolves.

hm thing sounds problemtatic, I didn't remember this

then better leave as is for now and maybe refactor in the future